core/src/packets.cpp

```diff
@@ -210,7 +210,7 @@ void AOClient::pktIcChat(AreaData *area, int argc, QStringList argv, AOPacket pa
         return;
     }
 
-    if (!server->isMessageAllowed()) {
+    if (!!area->isMessageAllowed() || !server->isMessageAllowed()) {
```

Comment:
This will cause no message to be accepted.
```suggestion
    if (!area->isMessageAllowed() || !server->isMessageAllowed()) {
```